src/groth16/params.rs

```diff
@@ -5,15 +5,24 @@ use crate::SynthesisError;
 use ec_gpu_gen::multiexp_cpu::SourceBuilder;
 
 use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
-use memmap::{Mmap, MmapOptions};
-use std::fs::File;
+
+#[cfg(feature = "memmap")]
+#[path = ""]
+pub mod memmap_uses {
+    pub use crate::groth16::MappedParameters;
+    pub use memmap::{Mmap, MmapOptions};
+    pub use std::fs::File;
+    pub use std::mem;
+    pub use std::ops::Range;
+    pub use std::path::PathBuf;
+}
 use std::io::{self, Read, Write};
-use std::mem;
-use std::ops::Range;
-use std::path::PathBuf;
 use std::sync::Arc;
 
-use super::{MappedParameters, VerifyingKey};
+#[cfg(feature = "memmap")]
+pub use memmap_uses::*;
```

Comment:
Wouldn't this make e.g. `File` public on `groth16::params`? I'd like to keep those imports private.

I would just add the `#[cfg(feature = "memmap")]` to all the imports that need it. It's quite ugly, but probably the easiest to reason about solution.

Reply:
The reason to use pub was to avoid warnings, but it seems it works fine now.